### ISSUE TITLE
Capture pc/server audio output e.g. from Spotify or Tidal

### ISSUE THREAD

**Issue by MoroseCorpse** [Splamy/TS3AudioBot#914]
I would like to intercept the audio output of my server. So that, for example, you play something via Spotify, Tidal or anything else and this is captured by the bot and output in TeamSpeak. 

I have to deal, although first with how I can, for example, Spotify on a Linux server only with console to run and this also plays something there. Because then you could just from his Spotify on the PC or mobile, control the music for the server, since you can choose the output device in Spotify or Tidal.

Does anyone know a way how I can implement this?

**Comment by Flakebi** [comment]
The easiest way would be to expose something as an http stream, which the audiobot already understands.

If that does not exist you probably want to look into pulseaudio and loopback devices. Then either expose a pulseaudio input device (“microphone”) as http stream, or write an audiobot plugin that catpures from a microphone.
Incidentally, the first versions of the audiobot worked liked that. vlc would play a song into a pulseaudio loopback device which redirected the sound into a real teamspeak client on the server.

If you don’t care about audiobot features, you could just run a real teamspeak client (xvfb is your friend for creating a virtual screen). However, it does use a lot more CPU than the audiobot, 30 % on a small VPS, compared to 1–2 % for the audiobot.

**Comment by MoroseCorpse** [comment]
Okay thanks. I tried to start Pulseaudio now, but I keep getting different errors. I know that Pulseaudio is not part of TS3AudioBot, but can you give me a short tutorial how you installed and started Pulseaudio on your servers? 

I have found the following bash code, which should then also directly create the http stream, but it fails already with the first commands
https://gist.github.com/djmaze/5234008

**Comment by Flakebi** [comment]
Starting pulseaudio should be as easy as running `pulseaudio` as the right user (i.e. the same use that will run spotify).
Do you see an error message?

**Comment by MoroseCorpse** [comment]
I already tried it with the command `pulseaudio`. I got the following error message: 
```
W: [pulseaudio] server-lookup.c: Unable to contact D-Bus: org.freedesktop.DBus.Error.NotSupported: Unable to autolaunch a dbus-daemon without a $DISPLAY for X11
W: [pulseaudio] main.c: Unable to contact D-Bus: org.freedesktop.DBus.Error.NotSupported: Unable to autolaunch a dbus-daemon without a $DISPLAY for X11
```

Also commands like `pulseaudio -D` or `pulseaudio --start` or `systemctl --user start pulseaudio` did not work. But by restarting the server sometimes a pulseaudio process was running. I found this out simply with the command: `ps -C pulseaudio`

But even if the process was supposedly running and I then tried to execute e.g. the first command from the already mentioned script, this error message came up: `No PulseAudio daemon running, or not running as session daemon.`
